game/views.py

```
@@ -33,12 +36,112 @@ def index(request):
     if 'game' not in request.session:
         game = ChessGame()
         request.session['game'] = game.to_dict()
-    return render(request, 'game/board.html')
+    return render(request, 'game/board.html', {
+        'activity_streak': get_activity_streak(request),
+    })
+
+
+def _calculate_activity_streak(dates, today=None):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_calculate_activity_streak`

(ANN202)